webserver.js

```diff
@@ -7261,41 +7259,306 @@ module.exports.CreateWebServer = function (parent, db, args, certificates, doneF
+                if (preset == 'google') { strategy.issuer.issuer = 'https://accounts.google.com'; }
+                if (!preset) {
+                    let error = new Error('OIDC: Missing issuer URI.');
+                    parent.authLog('error', `${error.message} STRATEGY: ${JSON.stringify(strategy)}`);
+                    throw error;
+                }
+            } else if ((typeof strategy.issuer.issuer == 'string') && (typeof strategy.custom.preset == 'string')) {
+                let error = new Error(`OIDC: PRESET: ${strategy.custom.preset.toUpperCase()}: PRESET OVERRIDDEN: CONFIG ISSUER: ${strategy.issuer.issuer} PRESET ISSUER: ${presetIssuer}`);
+                parent.authLog('setupDomainAuthStrategy', error.message);
+                console.warn(error)
+            }
+
+            // Setup Strategy Options
+            strategy.custom.scope = obj.common.convertStrArray(strategy.custom.scope, ' ')
+            if (strategy.custom.scope.length > 1) {
+                strategy.options = Object.assign(strategy.options, { 'params': { 'scope': strategy.custom.scope } })
+            } else {
+                strategy.options = Object.assign(strategy.options, { 'params': { 'scope': ['openid','profile','email'] } })
+            }
+            if (typeof strategy.groups == 'object') {
+                let groupScope = strategy.groups.scope || null
+                if (groupScope == null) {
+                    if (preset == 'azure') { groupScope = 'Group.Read.All' }
+                    if (preset == 'google') { groupScope = 'https://www.googleapis.com/auth/cloud-identity.groups.readonly' }
+                    if (typeof preset != 'string') { groupScope = 'groups' }
                 }
+                strategy.options.params.scope.push(groupScope)
             }
-        }
+            strategy.options.params.scope = strategy.options.params.scope.join(' ')
+
+            // Discover additional information if available, use endpoints from config if present
+            let issuer
+            try {
+                parent.authLog('setupDomainAuthStrategy', `OIDC: Discovering Issuer Endpoints: ${strategy.issuer.issuer}`);
+                issuer = await strategy.obj.openidClient.Issuer.discover(strategy.issuer.issuer);
+            } catch(err) {
+                let error = new Error('OIDC: Discovery failed.', { cause: err });
+                parent.authLog('setupDomainAuthStrategy', `ERROR: ${JSON.stringify(error)} ISSUER_URI: ${strategy.issuer.issuer}`);
+                throw error
+            } finally {
+                if (Object.keys(strategy.issuer).length > 1) {
+                    parent.authLog('setupDomainAuthStrategy', `OIDC: Adding Issuer Metadata: ${JSON.stringify(strategy.issuer)}`);
+                    issuer = new strategy.obj.openidClient.Issuer(Object.assign(issuer.metadata, strategy.issuer));
+                }
+                strategy.issuer = issuer.metadata
```

Comment:
This will fail as `issuer` is null if exception actually happened.

Reply:
If I remember correctly, getting issuer data here is needed and if this doesn't work the rest won't either. So if discovery fails and you get the error there is some misconfiguration, as in you manually entered an invalid issuer url while attempting to use a preset. I'd like to see your configs and logs for this if possible.

Reply:
It is caused by bad config, but the issue is that it should not fail on accessing `metadata` on undefined in successfully caught exception. If you want it in simples way, I would do: `            let issuer = {}`

Reply:
I removed the finally block